crates/bitcoind_rpc/src/lib.rs

```diff
@@ -91,44 +95,51 @@ impl<'c, C: bitcoincore_rpc::RpcApi> Emitter<'c, C> {
+            if tx_time > latest_time {
+                latest_time = tx_time;
+            }
 
-                    // Avoid emitting transactions that are already emitted if we can guarantee
-                    // blocks containing ancestors are already emitted. The bitcoind rpc interface
-                    // provides us with the block height that the tx is introduced to the mempool.
-                    // If we have already emitted the block of height, we can assume that all
-                    // ancestor txs have been processed by the receiver.
-                    let is_already_emitted = tx_time <= prev_mempool_time;
-                    let is_within_height = tx_entry.height <= prev_mempool_tip as _;
-                    if is_already_emitted && is_within_height {
-                        return None;
-                    }
+            // Avoid emitting transactions that are already emitted if we can guarantee blocks
+            // containing ancestors are already emitted. The bitcoind rpc interface provides us with
+            // the block height that the tx is introduced to the mempool. If we have already emitted
+            // the block of height, we can assume that all ancestor txs have been processed by the
+            // receiver.
+            let is_already_emitted = tx_time <= prev_mempool_time;
+            let is_within_height = tx_entry.height <= prev_mempool_tip as _;
+            if is_already_emitted && is_within_height {
+                continue;
+            }
 
-                    let tx = match client.get_raw_transaction(&txid, None) {
-                        Ok(tx) => tx,
+            match client.get_raw_transaction(&txid, None) {
+                Ok(tx) => {
+                    tx_update.txs.push(tx.into());
+                    tx_update.seen_ats.insert(txid, tx_time as u64);
+                }
+                Err(err) => {
+                    if err.is_not_found_error() {
                         // the tx is confirmed or evicted since `get_raw_mempool_verbose`
-                        Err(err) if err.is_not_found_error() => return None,
-                        Err(err) => return Some(Err(err)),
-                    };
-
-                    Some(Ok((tx, tx_time as u64)))
+                        continue;
+                    } else {
+                        return Err(err);
+                    }
                 }
-            })
-            .collect::<Result<Vec<_>, _>>()?;
+            };
+        }
 
         self.last_mempool_time = latest_time;
         self.last_mempool_tip = Some(self.last_cp.height());
 
-        Ok(txs_to_emit)
+        // After processing the mempool, any expected unconfirmed transactions that were not seen
+        // should be marked as evicted.
+        for expected_txid in expected_unconfirmed_txids {
+            if !tx_update.seen_ats.contains_key(&expected_txid) {
+                tx_update.evicted.insert(expected_txid);
+            }
+        }
```

Comment:
This is wrong as we skip including some mempool txs into `TxUpdate::txs` (check line 112).

Maybe create a `HashSet<Txid>` from the `.get_raw_mempool_verbose()` result directly to compare the `expected_unconfirmed_txids` against.